---
title: "ADM_2024_plastique"
output: pdf_document
date: "2024-01-11"
---

```{r}
library(dada2)
```


```{r}
path_to_fastqs <- "/Users/Anna/OneDrive/Cours/Master/M1/Semestre 8/stage/DADA2/dezippes"
# définir le chemin pour aller aux Fastas 
```

```{r}
list.files(path_to_fastqs)
```

```{r}
fnFs <- sort(list.files(path_to_fastqs,
                        pattern = "_1.fastq",
                        full.names = TRUE))
```

```{r}
fnRs <- sort(list.files(path_to_fastqs,
                        pattern = "_2.fastq",
                        full.names = TRUE))
```

```{r}
sample.names <- basename(fnFs) |>
  strsplit(split = "_") |> # diviser la chaîne de carac selon le modèle mis entre ""
  sapply(head, 1) # appliquer une fonction à chaque éléments d'une liste
sample.names
```

```{r}
basename(fnFs) |>
  strsplit(split = "_") |> # séparer les noms des fichiers en deux vecteurs
  head()
```

```{r}
plotQualityProfile(fnFs)
plotQualityProfile(fnRs)
```


```{r}
filtFs <- file.path(path_to_fastqs, "filtered", paste0(sample.names, "_1_filt.fastq"))
filtRs <- file.path(path_to_fastqs, "filtered", paste0(sample.names, "_2_filt.fastq"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names
```

```{r}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, trimLeft = c(20,20), truncLen=c(240,200),
                     maxN=0, maxEE = c(2,2),truncQ=2, rm.phix=TRUE,
                     compress=FALSE, multithread=FALSE)
head(out)
```

```{r}
errF <- learnErrors(filtFs, multithread=FALSE)

errR <- learnErrors(filtRs, multithread = FALSE)

plotErrors(errF, nominalQ=TRUE)
plotErrors(errR, nominalQ=TRUE)
```


```{r}
dadaFs <- dada(filtFs, err=errF, multithread=FALSE)

dadaRs <- dada(filtRs, err=errR, multithread=FALSE)


dadaFs[[1]]
```

```{r}
mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)

head(mergers[[1]])
```

```{r}
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
```

```{r}
table(nchar(getSequences(seqtab)))
```

```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=FALSE, verbose=TRUE)
dim(seqtab.nochim)
```

```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
head(track)
```

```{r}
taxa <- assignTaxonomy(seqtab.nochim, "C:/Users/Anna/OneDrive/Cours/Master/M1/Semestre 8/stage/DADA2/silva_nr99_v138.1_train_set.fa.gz", multithread=FALSE)

```

```{r}
taxa <- addSpecies(taxa, "C:/Users/Anna/OneDrive/Cours/Master/M1/Semestre 8/stage/DADA2/silva_species_assignment_v138.1.fa.gz")

```

```{r}
taxa.print <- taxa # Removing sequence rownames for display only
rownames(taxa.print) <- NULL
head(taxa.print)
```

```{r}
export_folder <- ("export")

if (!dir.exists(export_folder)) dir.create(export_folder, recursive = TRUE)

saveRDS(object = seqtab.nochim,
        file = file.path(export_folder, "seqtab.nochim.rds"))

saveRDS(object = taxa.print,
        file = file.path(export_folder, "taxa.rds"))
```

```{r}
write.csv(taxa.print, "tableASVfinale.csv")
```

